CIPs/CIP-23/CIP-23.md

```diff
@@ -0,0 +1,164 @@
+```
+cip: 23
+title: Social Accounts Definition
+author: Michael Sena (@michaelsena), Joel Thorstensson (@oed)
+status: Last Call
+category: Standards
+type: RFC
+created: 2020-07-21
+edited: 2020-11-03
+requires: Tile Doctype (CIP-8)
+```
+
+## Simple Summary
+
+**Social Accounts** is a data set that stores a list of a social accounts that are publicly linked to the users DID.
+
+
+## Abstract
+
+Users may want to publicly associate any number of social accounts, such as Twitter, GitHub, discord, etc. to their DID. Creating and storing public verifications for these various accounts in the Social Accounts *reference* document enables anyone to observe these associations and as a result infer trust about the given user. 
+
+Adding an account to the Social Accounts document *does not* allow that account to control or authenticate the DID. Rather it is simply just a verifiable public mapping.
+
+
+## Motivation
+
+Publicly associating one or more social accounts to a DID offers a number of benefits.
+
+- Public association with a DID to any number of accounts
+- Serves as a foundation for building aggregated reputation on an identity comprised of multiple accounts, not individual one-off accounts
+- Provides a common interface for creating a list of linked social accounts, making discoverability and query predictable
+
+
+## Specification
+
+The Social Accounts is a *definition* that contains an array of *linked accounts* entries.
+
+### Definition content
+
+**Deployment:** `<definition-DocID>`
+
+```json
+{
+  "name": "Social Accounts",
+  "description": "Social accounts linked to your DID",
+  "schema": "<reference-schema-DocID>",
+  "config": {
+    "tags": ["SocialAccounts"]
+  }
+}
+```
+
+### Reference Schema
+
+The reference schema defines a document which maintains an array of JSON objects that represent a DID's linked accounts. Below find the fields that are contained in each object.
+
+| Property       | Description                                                  | Value            | Max Size  | Required | Example                                      |
+| -------------- | ------------------------------------------------------------ | ---------------- | --------- | -------- | -------------------------------------------- |
+| `protocol`     | Network protocol used to resolve the host                    | string           | 50 char   | required | https                                        |
+| `host`         | Host used to resolve the ID                                  | string           | 150 char  | required | https://github.com                           |
+| `id`           | Path of the unique ID within the host. Can be the same as the host to represent the root. | string           | 450 char  | required | https://github.com/marysmith                 |
```

Comment:
What is the rationale between the `protocol`, `host` and `id` separation please?
It seems the `id` includes all the information provided by the other fields anyways. Maybe we could make it a `url` field with the associated validation requirement to be more explicit?

Reply:
I think it was made this way to accommodate for other types of social accounts on a variety of different protocols. This proposal was previously more broad. I guess we can simplify it. I think `url` makes sense.
Would be good to get input from @michaelsena here as well

Reply:
We made this general to support additional account types beyond Web2 social. In the case of a Matrix ID, or a Web3 domain name, you'd need additional data to specify where the registry is that might not fit into just a URL.

In the example of matrix:
Protocol: Matrix
Host: the location of your homeserver
ID: Your unique user ID on that homeserver (this is not an /append to the host as is the case with twitter)

In the example of ENS:
Protocol: Ethereum
Host: The chainID + contract address
ID: The user domain (or subdomain)

I'm wondering if we broaden this definition to be something like "Also Known As" which includes all of the linked accounts of the user. This is the terminology used by W3C in the DID spec nowadays.

With this naming though, I'm just wondering how we differentiate between Also Known As, and Crypto (CAIP-10) Accounts.

Reply:
I'm wondering if it makes sense to restrict this to mainly web2 accounts. Or alternatively merge it with crypto accounts (but that might cause more confusion).